grammars/jsdoc.cson

```diff
@@ -411,6 +430,92 @@
         ]
       }
     ]
+  # Default @param value, highlighted as JavaScript
+  'default-value':
+    'patterns': [
+      # Strings
+      {
+        'begin': '("|\')'
+        'beginCaptures':
+          '0':
+            'name': 'punctuation.definition.string.begin.js'
+        'end': '\\1|(?=\\*/)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.definition.string.end.js'
+        'name': 'string.quoted.js'
+        'patterns': [
+          {
+            'match': '\\\\(?:[^*]|\\*(?!/))'
+            'name': 'constant.character.escape.js'
+          }
+        ]
+      }
+      {
+        # Nested arrays matched to keep brackets balanced
+        'begin': '(\\[)'
+        'beginCaptures':
+          '0':
+            'name': 'source.embedded.js'
+          '1':
+            # NOTE: Should ideally be "punctuation.definition.bracket.square.begin.js"
+            # We're using the old/"incorrect" scope to ensure consistent highlighting.
+            'name': 'meta.brace.square.js'
+        'end': '(\\])|(?=\\*/)'
+        'endCaptures':
+          '0':
+            'name': 'source.embedded.js'
+          '1':
+            'name': 'meta.brace.square.js'
+        'patterns': [
+          {
+            'include': '#default-value'
+          }
+        ]
+      }
+      # Chunk of JS source (excluding nested arrays; see above). Note we "clamp" the
+      # imported highlighting within a fixed range to stop runaway problems with any
+      # begin/end patterns the JavaScript grammar matches.
+      {
+        'match': '(?:[^"\'\\[\\]*]|\\*[^/])+'
+        'name': 'source.embedded.js'
+        'captures':
+          '0':
+            'patterns': [
+              # Unmatched } after a string literal: `"string"}`
```

Comment:
Yeah.  So I didn't explain why because it was getting late last night but I still wanted to respond to let you know I got the notification.

Here's why I think this PR is unmergeable as-is, or even with an explanatory comment.
* These changes result in certain parts of language-javascript being spliced up such that the pattern begins matching but is interrupted before the end match.
* Therefore you had to add some out-of-context rules in an attempt to balance out the mismatched rules.  This is very prone to breakage as scopes or matches change.
* Inconsistent scopes.  Let's look at https://github.com/atom/language-javascript/pull/497/files#diff-c4166255a13782b6557ec1da0f4a6ef3R467 for an example.  Why is `"b"` not tokenized as embedded JavaScript, while it should be?  The whole default value should be embedded JS, yet there are portions when that scope disappears.  This of course is partly due to my insistence that double-matching should be avoided where possible, but that also brings me to my next point:
* Duplication of rules found in javascript.cson.  Again, this is very hard to keep in sync.
* Considerably, as you described it, "hairier" patterns for comparably small highlighting gains.

Overall, after reviewing this PR I am unconvinced that embedded Javascript highlighting for default values is worth it if it cannot be done in a clean manner that is easy to understand and requires minimal maintenance upkeep.